src/sparseml/core/lifecycle/session.py

```diff
@@ -155,6 +162,66 @@ def event(self, event_type: EventType, **kwargs) -> List[Any]:
 
         return mod_data
 
+    def _attach_logging_callbacks(self):
+        if (
+            isinstance(
+                self.state.loggers.frequency_manager, OptimizerStepFrequencyManager
+            )
+            and self.state.optimizer
+            and self.state.optimizer.optimizer
+        ):
+            optimizer = self.state.optimizer.optimizer
+            step_function = optimizer.step
+
+            def _optimizer_step_fn():
+                step_function()
+                current_step = int(
+                    optimizer.state[optimizer.param_groups[0]["params"][-1]][
+                        "step"
+                    ].item()
+                )
+                if should_log_model_info(
+                    model=self.state.model,
+                    loggers=self.state.loggers,
+                    current_log_step=current_step,
+                    last_log_step=self.state._last_log_step,
+                ):
+                    log_model_info(state=self.state, current_log_step=current_step)
+                    self.state._last_log_step = current_step
+
+            optimizer.step = _optimizer_step_fn
+
+        elif (
+            isinstance(self.state.loggers.frequency_manager, BatchFrequencyManager)
+            and self.state.model
+            and self.state.model.model
+        ):
+            model = self.state.model.model
+            forward_function = model.forward
```

Comment:
When possible we should try to avoid using PyTorch specific calls within core. A workaround would be to add a forward function to `ModifiableModel` and then call self.state.model.forward.

Reply:
Good Point, Noted will make that change